pkgs/development/tools/click/default.nix

```
@@ -0,0 +1,92 @@
+{ lib
+, fetchFromGitLab
+, buildPythonPackage
+, autoconf
+, debian
+, automake
+, perl
+, libtool
+, vala
+, pkg-config
+, libgee
+, json-glib
+, properties-cpp
+, gobject-introspection
+, getopt
+, setuptools
+, pygobject3
+, wrapGAppsHook
+}:
+
+buildPythonPackage {
+  pname = "click";
+  version = "unstable-2023-02-22";
+
+  src = fetchFromGitLab {
+    owner = "ubports";
+    repo = "development/core/click";
+    rev = "aaf2735e8e6cbeaf2e429c70136733513a81718a";
+    sha256 = "sha256-pNu995/w3tbz15QQVdVYBnWnAoZmqWj1DN/5PZZ0iZw=";
+  };
+
+
+  configureFlags = [
+    "--prefix=$out"
+    "--with-systemdsystemunitdir=$out/lib/systemd/system"
+    "--with-systemduserunitdir=$out/usr/lib/systemd/user"
+  ];
+
+  top_builddir = "$(pwd)";
+
+  click_cv_perl_vendorlib = "$out/lib";
+
+  preConfigure = ''
+    ./autogen.sh
+  '';
+
+  preFixup = ''
+    makeWrapperArgs+=(--prefix GI_TYPELIB_PATH : "$out/lib/click")
+  '';
```

`wrapGAppsHook` makes this obsolete, at the cost of a second layer of wrapping. What's still needed is setting `LD_LIBRARY_PATH`.
```suggestion
  preFixup = ''
    makeWrapperArgs+=(--prefix LD_LIBRARY_PATH : "$out/lib")
  '';
```